lib/Extractor/Candidates.cpp

```diff
@@ -860,6 +864,70 @@ std::tuple<BlockPCs, std::vector<InstMapping>> GetRelevantPCs(
   return std::make_tuple(RelevantBPCs, RelevantPCs);
 }
 
+void PrintDataflowInfo(Function &F, Instruction &I, LazyValueInfo *LVI,
+                       ScalarEvolution *SE) {
+  auto V = I.getOperand(0);
+  auto DL = F.getParent()->getDataLayout();
+  if (PrintNegAtReturn) {
+    bool Negative = isKnownNegative(V, DL);
+    if (Negative)
+      llvm::outs() << "known negative from compiler: " << "true" << "\n";
```

Comment:
don't keep duplicating strings like this, just print the first part of this before the conditional